### PR TITLE
Xeno blood is now finite

### PR DESCRIPTION
**Pull request by ThePiachu**
## About The Pull Request

While working on #1548 I got a comment how Xeno corpses produce infinite blood. That's obviously something that needs to be fixed if we want to do anything with Xeno blood products in the future (whether it's stims, exotic materials, equipment upgrades and so on). So I made the xeno blood actually tick down when you draw it out, and just to be sure I added a handle_blood() proc to xeno's Life() to make them regen blood while they are alive. You never know what ungas might do with a syringe...

Also fixed an edge case where you couldn't get the last drop of blood out of a mob if you were drawing too much. If that's the case we just take what's left.

## Why It's Good For The Game

If we want xeno blood to be useful for future development, it can't be a limitless resource.

## Changelog

:cl:
fix: Xeno blood is now a finite resource.
add: Living xenos now regenerate blood just in case some unga has the idea to attack one with a syringe.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/blood.dm [hunk 1]
```diff
@@ -171,6 +180,7 @@
 	for(var/plasma in plasmas)
 		O.reagents.add_reagent(plasma,amount / plasmas.len) //An even amount of each plasma and blood type
 
+	blood_volume = max(0, blood_volume - amount) // Removes blood if human
```
Comment (stanalbatross) [inline code/modules/mob/living/blood.dm:183]:
```suggestion
	blood_volume = max(0, blood_volume - amount)
```
this comment is wrong and so is its use in /carbon

File: code/modules/mob/living/blood.dm [hunk 2]
```diff
@@ -160,6 +166,9 @@
 	if(!O.reagents)
 		return
 
+	if(blood_volume < amount)
+		amount = blood_volume
```
Comment (stanalbatross) [inline code/modules/mob/living/blood.dm:170]: should probably sanity check for blood volume here so you don't go through the entire proc with `amount` being zero, same for /carbon

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/blood.dm [hunk 3]
```diff
@@ -132,11 +138,11 @@
 
 //Gets blood from mob to the container, preserving all data in it.
 /mob/living/carbon/proc/take_blood(obj/O, amount)
-	if(!O.reagents)
+	if(!O.reagents || amount <= 0)
```
Comment (stanalbatross) [inline code/modules/mob/living/blood.dm:141]:
```suggestion
	if(!O.reagents || amount <= 0 || blood_volume <= 0)
```

File: code/modules/mob/living/blood.dm [hunk 4]
```diff
@@ -157,9 +163,12 @@
 	. = ..()
 
 /mob/living/carbon/Xenomorph/take_blood(obj/O, var/amount)
-	if(!O.reagents)
+	if(!O.reagents || amount <= 0)
```
Comment (stanalbatross) [inline code/modules/mob/living/blood.dm:166]:
```suggestion
	if(!O.reagents || amount <= 0 || blood_volume <= 0)
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Review by fira — APPROVED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**